### PR TITLE
clang11: update libc++

### PR DESCRIPTION
**Pull request by ericLemanissier**
libc++-dev is version 6: https://packages.ubuntu.com/bionic/libc++-dev

Changelog: (Feature | Fix | Bugfix): Describe here your pull request

- [ ] Refer to the issue that supports this Pull Request.
- [ ] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [ ] I've read the [Contributing guide](https://github.com/conan-io/conan-docker-tools/blob/master/.github/CONTRIBUTING.md).
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008) style guides for Python code.
- [ ] I've followed the [Best Practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) guides for Dockerfile.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
I just realized it was already proposed in https://github.com/conan-io/conan-docker-tools/pull/252 but it looks like there was no newer docker image which came to fix the issue.
Having workarounds like this on CCI is very scary:
https://github.com/conan-io/conan-center-index/blob/1585b34/recipes/imagl/all/conanfile.py#L100-L101

**Comment by uilianries** [comment]
@ericLemanissier my concern about this PR is backward compatibility. I'm not sure if this change will break or not Conan packages already in Conan center and built by Clang 11 with libc++-6

**Comment by ericLemanissier** [comment]
according to https://releases.llvm.org/11.1.0/projects/libcxx/docs/DesignDocs/ABIVersioning.html it depends on cmake variable LIBCXX_ABI_VERSION used when building libc++. It is translated into c++ macro _LIBCPP_ABI_VERSION.
Current dockerio/clang11 has: `/usr/include/c++/v1/__config:#define _LIBCPP_ABI_VERSION 1`
With this PR we get `/usr/lib/llvm-11/include/c++/v1/__config:#  define _LIBCPP_ABI_VERSION 1` https://dev.azure.com/conanio/conan-docker-tools/_build/results?buildId=908&view=logs&j=e2666bbb-ec41-51a9-9241-02be2ffb2f19&t=9a0fad47-b207-55a8-8ea7-787dc35161c6&l=1889

File: clang_11/Dockerfile [hunk 1]
```diff
@@ -90,6 +90,10 @@ RUN dpkg --add-architecture i386 \
     && update-alternatives --install /usr/bin/python3 python3 /opt/pyenv/shims/python3 100 \
     && update-alternatives --install /usr/bin/pip pip /opt/pyenv/shims/pip 100 \
     && update-alternatives --install /usr/bin/pip3 pip3 /opt/pyenv/shims/pip3 100
+    
+RUN grep -r "LIBCPP_ABI_VERSION" /usr/ && \
+    echo "#include<algorithm>\
+    static_assert(_LIBCPP_ABI_VERSION==1);" | clang++ -stdlib=libc++ -std=c++17 -x c++ -c -
```
Comment (uilianries) [inline clang_11/Dockerfile:96]:
```suggestion
```

You can remove it.

**Review by uilianries — COMMENTED** [review]
I built Poco/1.11.0 and boost/1.76.0@ using this images, with all dependencies using the current libc++ version and worked.

**Review by uilianries — APPROVED** [review]

**Pull request merged**